File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        
        vector<int>arr = nums;
        
        for(int i = 0 ; i < nums.size(); i++){
            nums[(i+k)%nums.size()] = arr[i];
        }
    }
};

File: 0189-rotate-array/0189-rotate-array.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function rotate(nums: number[], k: number): void {
    let arr = [...nums];
    for(let i = 0; i<nums.length; i++){
        nums[(i+k)%nums.length] = arr[i];
    }
};